### ISSUE TITLE
Attachments not attached

### ISSUE THREAD

**Issue by pete-thompson** [spekt/junit.testlogger#40]
Forgive me if I've missed a configuration option for this, but I can't see any code to handle attachments so think I'm accurate.

I'm using MSTest and making use of the AddResultFile method to attach files to my test cases. I'd like those files to be included in the junit test results - it appears that the closest thing to a 'standard' is the Jenkins plugin which is also supported by GitLab: https://plugins.jenkins.io/junit-attachments/. 

Both GitLab and Jenkins support text written to stdout/stderr in a particular format - [[ATTACHMENT|/absolute/path/to/some/file]]. I'm not clear what represents an 'absolute path' in the context of CI/CD given that once files upload to the CI/CD servers the real absolute path isn't meaningful - my guess is that they mean a path relative to the base path for uploading result files.

**Comment by Siphonophora** [comment]
@pete-thompson  you are right this isn't something we currently are supporting. I just spent a few minutes looking at the docs, and its a little confusing. I guess GitLab is working with a different schema than we are. 

I think the absolute path is probably relative to the root of the repository. Probably not too hard to experiment. 

So I understand better, what is the distinction between an artifact and a attachment for you? If you were working in Gitlab CI/CD I know it would be easy to save all the files to a particular directory and then upload them all as artifacts, without relying on the attachment logic.

**Comment by pete-thompson** [comment]
Thanks @Siphonophora . I'm successfully uploading the files as artifacts, but I'd like them linked to the individual test cases in the JUnit report so that I can figure out which attachment goes with which test case.

In my particular situation I have a test case that's data driven - meaning that I'm running around 50 different tests under the same test name. I'm generating screenshots as attachments for each of the tests, which makes it hard to tell which file goes with which test instance. When I wrote all this, I was using Azure DevOps and the attachments were automatically linked in to the test reporting - but I'm now using GitLab which means I need to switch to JUnit. I'm going to update my tests to write out the filenames to the system output to make things a little easier, but it would be ideal if the JUnit report also contained the linkage.

Sadly, our on-premise GitLab instance doesn't support Junit report yet (the team that owns it were meant to upgrade it last week but delayed it) - so it's hard for me to test things yet, but I hope to write out log messages that would let me test the attachment feature as soon as our GitLab supports it.

**Comment by Siphonophora** [comment]
Was just looking at https://github.com/spekt/nunit.testlogger/issues/54 which is related. They would both depend on the core logger capturing attachment info to enable this feature.

**Comment by codito** [comment]
@Siphonophora we've added test attachments support in core testlogger with https://github.com/spekt/testlogger/commit/0a35c55a4249567213423c82903c1a8590f9807a. Here's the NUnit support for reference: https://github.com/spekt/nunit.testlogger/commit/577f53666262f0e474227053fbb2b63891eb5ccd.

We should consider updating version of JUnit logger to 3.1.x with this new feature.

**Comment by codito** [comment]
Leaving this link here for examples of adding attachment in junit xml: https://github.com/testmoapp/junitxml#attachments-screenshots-and-artifacts

**Comment by jdlegan** [comment]
Apologies for resurrecting this thread but was hoping to ask for some guidance. 

We are a .NET shop and utilize NUnit and are a Gitlab customer. We use JUnitXml.Testlogger 3.0.134 configured per your recommendation and Gitlabs own docs and it works great... until we tried to add attachments. No matter what I seem to do, I cannot get the necessary markup into the test results xml output. I thought it may be a matter of me simply using and extensibility point to insert the markup myself using a globally accessible object or method for the JUnitXml.Testlogger lib but I quickly reviewed the source of it and the core logger nothing jumped out at me.

What I have done per other issues in this project and in the NUnit Testlogger which is how I started pulling on this thread where support for attachments was discussed and I traced support back to the core logger for is:

- Make sure the attachments are available from the root of the CI Root in the pipeline (had originally be using temp files but moved them under the same directory as the test results xml themselves. 
- We are using the following command to run the tests: `dotnet test ./v${VERSION}.csproj --test-adapter-path:. --logger:"junit;LogFilePath=..\artifacts\sdktest\{assembly}-${VERSION}-test-result.xml;MethodFormat=Class;FailureBodyFormat=Verbose" --configuration Release
`
- We are adding attachments using `TestContext.AddTestAttachment(filename, description)` where filename is accessible (bullet 1).
- Using NUnit 4.0.1 (.NET 8 but same behavior was observed in .NET 7).
- Attachments work as expected in VS using MSTest (attachments included in test results, clickable, view contents).

I would think the easiest solution would be to figure out how to get the attachment detail into the resultant XML before its written, either because it is handled by the JUnitXML or core logger OR because it was manually inserted but would love to avoid doing it as a post-processing of the files after they are written but before the stage completes and Gitlab parses the results. 

Does anyone have any pointers on how to achieve this?
